Replace all chevrons with a single icon

jryans opened the pull request:
This replaces all chevrons with a single icon, affecting components like the
room list, network dropdown, and rebrand modal.

Rather than having files for each direction that can get out of sync, this
change uses one direction and rotates in CSS for the others.

## Network dropdown

<img width="266" alt="image" src="https://user-images.githubusercontent.com/279572/89059909-f7bb0380-d359-11ea-8cc8-44dd0e51149d.png">

## Rebrand modal

<img width="581" alt="image" src="https://user-images.githubusercontent.com/279572/88949911-cf64d380-d28b-11ea-92ec-04b830bac389.png">

## Room list

<img width="287" alt="image" src="https://user-images.githubusercontent.com/279572/89057877-45ce0800-d356-11ea-8391-8a2ac0009e11.png">

<img width="281" alt="image" src="https://user-images.githubusercontent.com/279572/89057921-53838d80-d356-11ea-9124-24f2f775b587.png">

## Jump to bottom

<img width="63" alt="image" src="https://user-images.githubusercontent.com/279572/89062001-99901f80-d35d-11ea-81e9-8fdaff649ba9.png">

## Jump to unread

<img width="62" alt="image" src="https://user-images.githubusercontent.com/279572/89070814-7a00f300-d36d-11ea-8713-8db6aad175b6.png">

## With guides to confirm alignment

<img width="68" alt="image" src="https://user-images.githubusercontent.com/279572/89071298-4e323d00-d36e-11ea-89bf-9ce3d5c9d01f.png">

Fixes https://github.com/vector-im/riot-web/issues/14663

benedikrok commented:
@jryans 
- Show more should have the same chevron as "People" title for example
- The network dropdown can be scaled up a bit and repositioned so it's vertically centered to the "X" icon above, and we can nudge it down a bit so it's horizontally aligned with the text 'now it's a it up)
- Riot to Element is a surprise to me, since it's more of an illustrative composition than a regular UI. If we are re-using the same icon in this case, make it bigger and probably slightly thicker

jryans commented:
> * Show more should have the same chevron as "People" title for example

Hmm, so the chevron next to People and next to Show more should be the same icon at the same size... Do you want both to be the same colour then? If so, which colour, or have I misunderstood entirely...? 😅 

> * The network dropdown can be scaled up a bit and repositioned so it's vertically centered to the "X" icon above, and we can nudge it down a bit so it's horizontally aligned with the text 'now it's a it up)

Okay, I have made it larger to match the X icon above, and repositioned to align with text and X icon. Screenshot updated above.

> * Riot to Element is a surprise to me, since it's more of an illustrative composition than a regular UI. If we are re-using the same icon in this case, make it bigger and probably slightly thicker

Yeah, it just happened to be using the chevron icon previously, and I didn't really want to leave it around just for that screen, so hoping to adapt the new one to fit here as well. I made it bigger and thicker, screenshot updated above.

benedikrok commented:
> > * Show more should have the same chevron as "People" title for example
> 
> Hmm, so the chevron next to People and next to Show more should be the same icon at the same size... Do you want both to be the same colour then? If so, which colour, or have I misunderstood entirely...? 😅
> 

Here is an example of how do they look like in context: https://www.figma.com/file/rHnynEgrePUtXVear8rGT0/34-Room-List?node-id=657%3A12676
Same size, same color (both light and dark theme), just different positioning 


> > * The network dropdown can be scaled up a bit and repositioned so it's vertically centered to the "X" icon above, and we can nudge it down a bit so it's horizontally aligned with the text 'now it's a it up)
> 
> Okay, I have made it larger to match the X icon above, and repositioned to align with text and X icon. Screenshot updated above.
> 

Perfect sizing! I'd just move down a bit more and match the thickness of the lines with the "X" above.

> > * Riot to Element is a surprise to me, since it's more of an illustrative composition than a regular UI. If we are re-using the same icon in this case, make it bigger and probably slightly thicker
> 
> Yeah, it just happened to be using the chevron icon previously, and I didn't really want to leave it around just for that screen, so hoping to adapt the new one to fit here as well. I made it bigger and thicker, screenshot updated above.

Looks great!

jryans commented:
> Here is an example of how do they look like in context: https://www.figma.com/file/rHnynEgrePUtXVear8rGT0/34-Room-List?node-id=657%3A12676
> Same size, same color (both light and dark theme), just different positioning

Hmm, I think I am missing whether that's what you want or what is wrong... As far as I can tell, the size and position for both matches the Figma screen. The colour is currently different. Should I change the color to match Figma? Or is something wrong with the position? If so, you may need to be more verbose, as I can't tell what is different from Figma at the moment. 😅

benedikrok commented:
> > Here is an example of how do they look like in context: https://www.figma.com/file/rHnynEgrePUtXVear8rGT0/34-Room-List?node-id=657%3A12676
> > Same size, same color (both light and dark theme), just different positioning
> 
> Hmm, I think I am missing whether that's what you want or what is wrong... As far as I can tell, the size and position for both matches the Figma screen. The colour is currently different. Should I change the color to match Figma? Or is something wrong with the position? If so, you may need to be more verbose, as I can't tell what is different from Figma at the moment. 😅

The show more seems smaller for some reason in the preview. The alignment looks fine, but the colors should be different.
I don't have a clue why colors are different but the Room sections, Show more and chevrons on both should be in the same color.

jryans commented:
> The show more seems smaller for some reason in the preview. The alignment looks fine, but the colors should be different.
> I don't have a clue why colors are different but the Room sections, Show more and chevrons on both should be in the same color.

Thanks for clarifying. I have double-checked to verify, and the chevrons for room section and show more are the same size.

As for colours, there's actually several colours in Figma as well, so I am not sure which you want:

## Figma

* Both chevrons, room list section header text: #8D99A5 (tertiary text)
* Show more text: #737D8C (secondary text)

The same is true in the dark theme version: show more text uses secondary, while the others use tertiary.

## Current app without changes

### Light theme

* Show more text and chevron: #737D8C (secondary text)
* Room list section header text and chevron: #2e2f32 (primary fg text)

### Dark theme

* Show more text and chevron: #A9B2BC (room tile preview)
* Room list section header text: #8E99A4 (only used here) 
* Room list section header chevron: #ffffff (primary fg text)

@benedikrok, should everything be changed to tertiary text colours in both themes then...? Does that include the message previews on room tiles as well...?

benedikrok commented:
> > The show more seems smaller for some reason in the preview. The alignment looks fine, but the colors should be different.
> 
> > I don't have a clue why colors are different but the Room sections, Show more and chevrons on both should be in the same color.
> 
> 
> 
> Thanks for clarifying. I have double-checked to verify, and the chevrons for room section and show more are the same size.
> 
> 
> 
> As for colours, there's actually several colours in Figma as well, so I am not sure which you want:
> 
> 
> 
> ## Figma
> 
> 
> 
> * Both chevrons, room list section header text: #8D99A5 (tertiary text)
> 
> * Show more text: #737D8C (secondary text)
> 
> 
> 
> The same is true in the dark theme version: show more text uses secondary, while the others use tertiary.
> 
> 
> 
> ## Current app without changes
> 
> 
> 
> ### Light theme
> 
> 
> 
> * Show more text and chevron: #737D8C (secondary text)
> 
> * Room list section header text and chevron: #2e2f32 (primary fg text)
> 
> 
> 
> ### Dark theme
> 
> 
> 
> * Show more text and chevron: #A9B2BC (room tile preview)
> 
> * Room list section header text: #8E99A4 (only used here) 
> 
> * Room list section header chevron: #ffffff (primary fg text)
> 
> 
> 
> @benedikrok, should everything be changed to tertiary text colours in both themes then...? Does that include the message previews on room tiles as well...?

You're right, I forgot we made an exception for the show more text. Let's stick with the values we have in Figma. Sorry for the confusion 😬

turt2live commented on res/css/views/directory/_NetworkDropdown.scss:
```diff
@@ -147,7 +147,8 @@ limitations under the License.
         width: 24px;
         height: 24px;
-        right: -28px; // - (24 + 4)
+        right: -26.5px; // - (24 + 2.5)
```
while we're here, can we reference what 24 and 2.5 mean?

bwindels commented:
Nique was asking me to also replace the jump down chevron with this one as well during our design polish session, could you also use the new chevron here please :pray:?
```
.mx_JumpToBottomButton_scrollDown::before {
    content: "";
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    mask: url('$(res)/img/icon-jump-to-bottom.svg');
    mask-repeat: no-repeat;
    mask-position: center;
    background: $muted-fg-color;
}
```

benedikrok commented:
> Nique was asking me to also replace the jump down chevron with this one as well during our design polish session, could you also use the new chevron here please 🙏?
> 
> ```
> .mx_JumpToBottomButton_scrollDown::before {
>     content: "";
>     position: absolute;
>     top: 0;
>     bottom: 0;
>     left: 0;
>     right: 0;
>     mask: url('$(res)/img/icon-jump-to-bottom.svg');
>     mask-repeat: no-repeat;
>     mask-position: center;
>     background: $muted-fg-color;
> }
> ```

How does it look like?

bwindels commented:
> > Nique was asking me to also replace the jump down chevron with this one as well during our design polish session, could you also use the new chevron here please pray?
> > ```
> > .mx_JumpToBottomButton_scrollDown::before {
> >     content: "";
> >     position: absolute;
> >     top: 0;
> >     bottom: 0;
> >     left: 0;
> >     right: 0;
> >     mask: url('$(res)/img/icon-jump-to-bottom.svg');
> >     mask-repeat: no-repeat;
> >     mask-position: center;
> >     background: $muted-fg-color;
> > }
> > ```
> 
> How does it look like?

See https://github.com/matrix-org/matrix-react-sdk/pull/5074

benedikrok commented:
> > > Nique was asking me to also replace the jump down chevron with this one as well during our design polish session, could you also use the new chevron here please pray?
> 
> > > ```
> 
> > > .mx_JumpToBottomButton_scrollDown::before {
> 
> > >     content: "";
> 
> > >     position: absolute;
> 
> > >     top: 0;
> 
> > >     bottom: 0;
> 
> > >     left: 0;
> 
> > >     right: 0;
> 
> > >     mask: url('$(res)/img/icon-jump-to-bottom.svg');
> 
> > >     mask-repeat: no-repeat;
> 
> > >     mask-position: center;
> 
> > >     background: $muted-fg-color;
> 
> > > }
> 
> > > ```
> 
> > 
> 
> > How does it look like?
> 
> 
> 
> See https://github.com/matrix-org/matrix-react-sdk/pull/5074

Let's change it. Just make sure the sizing will make sense. I. It default size it might be too small, so we should probably resize it.

bwindels commented:
And as well for the top unread button in the timeline please?

```
mx_TopUnreadMessagesBar_scrollUp::before {
    ...
    mask-image: url('$(res)/img/icon-jump-to-first-unread.svg');
    ...
}
```

benedikrok commented:
> And as well for the top unread button in the timeline please?
> 
> ```
> mx_TopUnreadMessagesBar_scrollUp::before {
>     ...
>     mask-image: url('$(res)/img/icon-jump-to-first-unread.svg');
>     ...
> }
> ```

👍

jryans commented:
@benedikrok Today's updates include:

* Room list colours should now match Figma
* Network dropdown chevron is thinner, moved down slightly to center with adjacent text
* Converted jump to bottom button
* Converted jump to unread button

See updated screenshots in the top comment to check the results.

benedikrok commented:
Looks Good 👍

benedikrok commented:
> @benedikrok Today's updates include:
> 
> * Room list colours should now match Figma
> * Network dropdown chevron is thinner, moved down slightly to center with adjacent text
> * Converted jump to bottom button
> * Converted jump to unread button
> 
> See updated screenshots in the top comment to check the results.

Looks Good 👍

jryans commented:
Thanks all! 😄
